azalea-auth/src/auth.rs

```diff
@@ -280,25 +288,27 @@ pub enum GetMicrosoftAuthTokenError {
 ///
 /// ```
 /// # async fn example(client: &reqwest::Client) -> Result<(), Box<dyn std::error::Error>> {
-/// let res = azalea_auth::get_ms_link_code(&client).await?;
+/// let res = azalea_auth::get_ms_link_code(&client, "client_id", "scope").await?;
 /// println!(
 ///     "Go to {} and enter the code {}",
 ///     res.verification_uri, res.user_code
 /// );
-/// let msa = azalea_auth::get_ms_auth_token(client, res).await?;
+/// let msa = azalea_auth::get_ms_auth_token(client, res, "client_id").await?;
 /// let minecraft = azalea_auth::get_minecraft_token(client, &msa.data.access_token).await?;
 /// let profile = azalea_auth::get_profile(&client, &minecraft.minecraft_access_token).await?;
 /// # Ok(())
 /// # }
 /// ```
 pub async fn get_ms_link_code(
     client: &reqwest::Client,
+    client_id: &str,
+    scope: &str
```

Comment:
I think this should take `Option<&str>`s to make it like some of the other functions and so the user isn't required to pass a client id and scope.
```suggestion
    client_id: Option<&str>,
    scope: Option<&str>
```